src/commands/selectCommandTemplate.ts

```diff
@@ -0,0 +1,157 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/* eslint-disable no-template-curly-in-string */
+
+import * as vscode from 'vscode';
+import { IActionContext, IAzureQuickPickItem } from 'vscode-azureextensionui';
+import { ext } from '../extensionVariables';
+import { resolveVariables } from '../utils/resolveVariables';
+
+export type TemplateCommand = 'build' | 'run' | 'runInteractive' | 'attach' | 'logs' | 'composeUp' | 'composeDown';
+
+type CommandTemplate = {
+    template: string,
+    label: string,
+    match?: string,
+};
+
+// NOTE: the default templates are duplicated in package.json, since VSCode offers no way of looking up extension-level default settings
+// So, when modifying them here, be sure to modify them there as well!
+const defaults: { [key in TemplateCommand]: CommandTemplate } = {
+    'build': { label: 'Docker Build', template: 'docker build --rm -f "${dockerfile}" -t ${tag} "${context}"' },
+    'run': { label: 'Docker Run', template: 'docker run --rm -d ${exposedPorts} ${tag}' },
+    'runInteractive': { label: 'Docker Run (Interactive)', template: 'docker run --rm -it ${exposedPorts} ${tag}' },
+    'attach': { label: 'Docker Attach', template: 'docker exec -it ${containerId} ${shellCommand}' },
+    'logs': { label: 'Docker Logs', template: 'docker logs -f ${containerId}' },
+    'composeUp': { label: 'Compose Up', template: 'docker-compose up ${configurationFile} ${detached} ${build}' },
+    'composeDown': { label: 'Compose Down', template: 'docker-compose down ${configurationFile}' },
+};
+
+export async function selectBuildCommand(context: IActionContext, matchContext: string[], folder: vscode.WorkspaceFolder, dockerfile: string, buildContext: string): Promise<string> {
+    return await selectCommandTemplate(
+        context,
+        'build',
+        matchContext,
+        folder,
+        { 'dockerfile': dockerfile, 'context': buildContext }
+    );
+}
+
+export async function selectRunCommand(context: IActionContext, matchContext: string[], tag: string, interactive: boolean, exposedPorts?: { [portAndProtocol: string]: {} }): Promise<string> {
+    let portsString: string = '';
+    if (exposedPorts) {
+        portsString = Object.keys(exposedPorts).reduce((partialPortsString: string, portAndProtocol: string) => {
+            return `${partialPortsString} -p ${portAndProtocol.split('/')[0]}:${portAndProtocol}`
+        }, portsString);
+    }
+
+    return await selectCommandTemplate(
+        context,
+        interactive ? 'runInteractive' : 'run',

```

Comment:
Do we care about the "run Azure CLI" flavor?

Reply:
I didn't implement it in settings either; I'm not sure why users would need to customize it...

Reply:
Agree it's not a P1 scenario, but I can imagine that users may want to, for example, volume mount some directory full of ARM templates/scripts.

Reply:
I guess a potential workaround would be to pull the Azure CLI image and define a custom Run Interactive command scoped to that image. You would have to go through the explorer / command palette + select, rather than just command palette, but it's something users will be able to do immediately upon release.